picard/ui/scriptsmenu.py

```diff
@@ -66,20 +66,19 @@ def _run_script(self, script):
                 }
                 self.tagger.window.set_statusbar_message(msg, mparms)
 
-    def _get_unique_metadata_objects(self):
-        objs = self._get_metadata_objects(self.tagger.window.selected_objects)
-        return uniqify(objs)
+    def _iter_unique_metadata_objects(self):
+        return iter_unique(self._iter_metadata_objects(self.tagger.window.selected_objects))
 
-    def _get_metadata_objects(self, objs):
+    def _iter_metadata_objects(self, objs):
         for obj in objs:
             if hasattr(obj, 'metadata'):
                 yield obj
             if isinstance(obj, Cluster):
-                yield from self._get_metadata_objects(obj.files)
+                yield from self._iter_metadata_objects(obj.iterfiles())
```

Comment:
Wouldn't it be more correct to have `elif` instead of `if` here?

Reply:
I changed it to elif for the checks below. Note that the `hasattr(obj, 'metadata'):` is separate in addition to the rest (yield the object itself, yield it's children).